crates/ide-assists/src/handlers/qualify_method_call.rs

```diff
@@ -281,7 +281,7 @@ use test_mod::*;
 
 fn main() {
     let test_struct = test_mod::TestStruct {};
-    TestTrait::test_method(&test_struct)
```

Comment:
This isn't right, we want this to stay the trait here.
The code responsible here is https://github.com/rust-lang/rust-analyzer/blob/434e718b6717246066898a2dfcda1aec5e6f4082/crates/ide-assists/src/handlers/qualify_method_call.rs#L50-L51, we might need to copy the `item_for_path_search` function and special case the `impl` case to take the trait for this assist